src/main/java/duke/task/Deadline.java

```diff
@@ -0,0 +1,39 @@
+package duke.task;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * A Deadline class which represents a deadline task.
+ */
+public class Deadline extends Task {
+    protected LocalDate by;
+
+    /**
+     * Constructs Deadline.
+     *
+     * @param isDone      Determines if task is completed.
+     * @param description Task description.
+     * @param by          Due date.
```

Comment:
Perhaps renaming `by` to `dueDate` would be more descriptive? Same for the `Event` class.